### PR TITLE
MAINTAINERS: Add Infineon Platforms section

### PR DESCRIPTION
**Pull request by ifyall**
Add the Infineon Platforms section, with maintainer, contributors and appropriate file patterns.

### CODE REVIEW

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -2339,6 +2339,40 @@ Xilinx Platforms:
...
+    - ifyall
+  collaborators:
+    - npal-cy
+    - talih0
+    - billwatersiii
+    - sreeramIfx
+    - dey-r
+    - mcatee-infineon
+  files:
+    - boards/arm/nucleo_*/
```
Comment (mbolivar-nordic) [inline MAINTAINERS.yml:2354]: This isn't right; is it?

Comment (ifyall) [reply MAINTAINERS.yml:2354]: Nope, that's copy-paste biting once again.

**Review by mbolivar-nordic — DISMISSED** [review]

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -2339,6 +2339,36 @@ Xilinx Platforms:
...
+  status: maintained
+  maintainers:
+    - ifyall
+  collaborators:
+    - npal-cy
+    - talih0
+    - billwatersiii
+    - sreeramIfx
+    - dey-r
+    - mcatee-infineon
```
Comment (stephanosio) [inline MAINTAINERS.yml:2352]: These users do not seem to have made any contributions to the project yet, and [a "collaborator" needs to be a "contributor" by definition](https://docs.zephyrproject.org/latest/project/project_roles.html).

**Review by stephanosio — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml (hunk 2 above)
Comment (ifyall) [reply MAINTAINERS.yml]: Noted and removed.

**Review by stephanosio — DISMISSED** [review]

**Review by galak — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by galak** [comment]
@ifyall please rebase, no merge commits.

**Review by MaureenHelm — DISMISSED** [review]

**Review by stephanosio — DISMISSED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Pull request merged**